core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -236,26 +246,62 @@ newtype GTxId = GTxId (Hash GTx)
 
 -- | Compute tx id.
 toGTxId :: (Serialise Tx, Serialise PublicationTx) => GTx -> GTxId
-toGTxId (GMoneyTx tx) = GTxId . unsafeCastHash . toTxId $ tx
+toGTxId (GMoneyTx tx)       = GTxId . unsafeCastHash . toTxId $ tx
 toGTxId (GPublicationTx tx) = GTxId . unsafeCastHash . toPtxId $ tx
 
-data GTxWitnessed
-    = GMoneyTxWitnessed TxWitnessed
+data CarriedKind = Size | Fee
```

Comment:
ELI5, why is fee and size stored separately?

Also, as far as I see, we now store tx with size (not fee, see `GTxWitnessed`) in block, is this intended?
When I have a block with transactions (e.g. in block explorer), how do I get the fee for a particular transactions?

Reply:
Below there is `turnIntoFees` function that performs `Carrying 'Size GTxwitnessedRaw -> Carrying 'Fee GTxwitnessedRaw`.

The reason I did it like that is because size is only needed to calculate fees, and reason to calculated fee only emerges in (Tx -> SD.Tx) expander, where I use `turnIntoFees`.

Fee and size represent different stages of fee calculation and they are stored disjunctively, not separately. The whole `CarriedKind` thing is so you can't calculate fees twice or zero times.

Reply:
I, for one, don't see any particular use for a transaction with size _and_ without fee.
Why not calculate and store them both at the same time?

And, once again, how do I reliably get the transaction fee from `GTxWitnessed` in block explorer?

Reply:
Moreover, there is a use for a function, which directly calculates fee for a transaction.
It will be needed in wallet app, when we show fee preview for outgoing transactions.